### ISSUE TITLE
failed to initialize cache: redis: invalid redis URL path: /harbor

### ISSUE THREAD

**Issue by dmitry-g** [goharbor/harbor#17483]
Hello, when switching docker-compose based Harbor from v2.5.3 to v2.6.0 we faced the following error:

```
harbor-core    | 2022-08-31T13:51:36Z [INFO] [/core/main.go:136]: initializing cache ..
harbor-core    | 2022-08-31T13:51:36Z [FATAL] [/core/main.go:138]: failed to initialize cache: redis: invalid redis URL path: /harbor
```

**Steps to reproduce the problem:**

Create harbor.yml with external redis sentinel configuration, run prepare script and start docker-compose application (see _Additional context_)

**Versions:**

- harbor version: 2.6.0
- docker engine version: 19.03.15, build 99e3ed8919
- docker-compose version: 1.29.2, build 5becea4c

**Additional context:**

harbor.yml

```yaml
# [...]
external_redis:
  host: <redis_ip>:16379
  sentinel_master_set: harbor
  # db_index 0 is for core, it's unchangeable
  registry_db_index: 1
  jobservice_db_index: 2
  chartmuseum_db_index: 3
  trivy_db_index: 5
  idle_timeout_seconds: 30
# [...]
cache:
  enabled: false
  expire_hours: 24
```

Generated common/config/core/env

```shell
# [...]
_REDIS_URL_CORE=redis+sentinel://<redis_ip>:16379/harbor?idle_timeout_seconds=30
_REDIS_URL_REG=redis+sentinel://<redis_ip>:16379/harbor/1?idle_timeout_seconds=30
# [...]
```

**Problem**

`prepare` scripts generates the path of `_REDIS_CORE_URL` without db index (for example, `/harbor` instead of `/harbor/0`)

db index 0 is reserved for core:
https://github.com/goharbor/harbor/blob/b035ca7cd9c3547793fef7c0aec858c04ce4f8a6/make/photon/prepare/utils/configs.py#L456

When `db` argument is 0, the following line always evaluates to `""`:

https://github.com/goharbor/harbor/blob/b035ca7cd9c3547793fef7c0aec858c04ce4f8a6/make/photon/prepare/utils/configs.py#L389

On the other hand, when core initializes cache, it requires `_REDIS_CORE_URL` to contain exactly 2 parts in its path - master name and db index:

https://github.com/goharbor/harbor/blob/b035ca7cd9c3547793fef7c0aec858c04ce4f8a6/src/lib/cache/redis/util.go#L44-L50

Thus it returns an error while trying to parse the URL generated by `prepare` scripts.

**Solution**

I would like to raise a PR with a fix, but first I would like to hear your opinion on which approach to choose here:

 - Generate `_REDIS_CORE_URL` which would contain `0` db index
 - Change validation of sentinel URL for cache layer

Thanks

**Comment by chlins** [comment]
@dmitry-g Thanks! It's the bug imported in v2.6.0, and I'll fix this issue in the v2.6.1.

**Comment by chlins** [comment]
This is a known issue in the harbor v2.6.0, and will be fixed in the v2.6.1. Here is the workaround, update the redis URL environment variable for the core, add the database index 0 in the URL.

`_REDIS_URL_CORE=redis+sentinel://<redis_ip>:16379/harbor?idle_timeout_seconds=30` => `_REDIS_URL_CORE=redis+sentinel://<redis_ip>:16379/harbor/0?idle_timeout_seconds=30`
